### DISCUSSION TITLE
文章修正提案機能の指摘がちょっとstrictすぎるかもしれない

### DISCUSSION THREAD

**Discussion by JunichiIto** [increments/qiita-discussions#701] [要望・アイデア・不具合]
[【RSpec】フレーキーなテスト（たまに落ちるテスト）の直し方 \#Ruby \- Qiita](https://qiita.com/jnchito/items/56ba52c9a7246fc5016a)

上記記事を書いてる途中で、文章修正提案機能が赤線を引いてきたものの、「それぐらいは許容範囲なのでは？」と思ったものをいくつかリストアップします。

![Screenshot 2024-04-09 at 9 25 47](https://github.com/increments/qiita-discussions/assets/1148320/5bce575a-9759-4609-9149-4bfdc13465eb)
↑「Single Page Application(SPA)」って書いてほしい、ということでしょうか？
でも「略語（略さない名称）」で書きたいときもあると思うんですよね。

![Screenshot 2024-04-09 at 9 26 31](https://github.com/increments/qiita-discussions/assets/1148320/3c27b004-1868-41d5-9e66-c51149f2aa2b)
↑これぐらいええやん。。。っていうお気持ち。

![Screenshot 2024-04-09 at 9 26 21](https://github.com/increments/qiita-discussions/assets/1148320/776cfd81-7f0d-4907-8cd3-04464659a1c5)
↑同じくこれぐらいええやん。。。っていうお気持ち。

![Screenshot 2024-04-09 at 9 26 12](https://github.com/increments/qiita-discussions/assets/1148320/15ac8aa8-fc28-410d-b70b-c195c416bec1)
↑上に同じ

![Screenshot 2024-04-09 at 9 26 09](https://github.com/increments/qiita-discussions/assets/1148320/ccded572-f4dc-41ba-985c-773466830061)
↑上に同じ

![Screenshot 2024-04-09 at 9 25 58](https://github.com/increments/qiita-discussions/assets/1148320/76b94288-e8db-4437-8bfb-1c50de12eb78)
↑上に同じ

助詞の連続はなるべく避けた方がいいとは思いますが、MUSTではないと思いますし、それ以外の言い換え方もない（無理に言い換えるとかえって不自然になる）ことも多いので、「そのままにさせてよ」と思いました。
指摘されても無視すればよい、という考え方もあるかもしれませんが、赤線が引かれているのは気持ち悪いですし、ブラウザ本体のスペルチェックとも似ているのでどうしても「修正しなければならない問題」に見えてしまいます。

![Screenshot 2024-04-09 at 9 47 22](https://github.com/increments/qiita-discussions/assets/1148320/5e85ee76-40fe-412c-bfdd-43adf503d237)

なので、明らかに修正が必要な問題（主語と述語の係り受けがおかしい場合など）以外は、指摘しないでほしいなーと思いました。

一意見として参考にしていただけると幸いです。

**Comment by tanaka-asami415** [comment]
@JunichiIto 
ご意見ありがとうございます！
直近の開発優先度などからすぐの対応は難しいと思いますが、指摘を緩和する or 文章修正提案機能や指摘の [ルール](https://qiita.com/release-notes/2023-12-08) 毎にオン・オフをできるようにするなどを検討したいと思います！

**Reply by JunichiIto** [reply]
返信ありがとうございます。ご検討よろしくお願いします。

P.S.
Wordの「文章校正の詳細設定の画面」みたいなのがあると、僕みたいなユーザーはありがたいかも。
（開発工数は無視してますが😅）

https://kousei.club/word%e3%81%a7%e6%96%87%e7%ab%a0%e3%81%ae%e6%a0%a1%e6%ad%a3%e6%a0%a1%e9%96%b2%e6%96%87%e7%ab%a0%e6%a0%a1%e6%ad%a3%e6%a9%9f%e8%83%bd%e3%82%92%e3%83%95%e3%83%ab%e3%81%ab%e6%b4%bb%e3%81%8b/

**Comment by JunichiIto** [comment]
以下は気になった警告に関する追記です。（返信不要）

**Reply by JunichiIto** [reply]
![Screenshot 2024-04-14 at 11 17 36](https://github.com/increments/qiita-discussions/assets/1148320/bf71d3b9-935c-4bb8-8139-3cf39e55c2c3)

これは別に間違った用法ではないのでは？

**Reply by JunichiIto** [reply]
![Screenshot 2024-04-15 at 8 23 55](https://github.com/increments/qiita-discussions/assets/1148320/9c890e64-0cb2-4c9e-a5ae-ddede43b759a)

三点リーダはむしろ2回連続で使う方が正しいです。（と、チェリー本を執筆してるときに編集者さんに教えてもらった）

> このように、文法上は三点リーダーを並べて使うのが正しい使い方とされています
> https://estlinks.co.jp/column/leader/

**Reply by JunichiIto** [reply]
![Screenshot 2024-04-15 at 9 14 26](https://github.com/increments/qiita-discussions/assets/1148320/5e9b6e23-18b5-4086-9c5a-5437dc3cd362)

「話題の広がりすぎる」じゃなくて「話題が広がりすぎる」でしょ！みたいなツッコミがほしかったけど、ここは指摘してもらえなかった。

**Reply by JunichiIto** [reply]
![Screenshot 2024-04-29 at 19 35 11](https://github.com/increments/qiita-discussions/assets/1148320/a2ec9b19-1c8a-41f3-8435-c6f24c15e3fb)

ビックリマークを連続させるぐらい許可してよ、というお気持ち。

**Reply by P-SiZK** [reply]
@JunichiIto
複数のご報告をいただきありがとうございました！
連続する単語での修正提案機能のルールを緩和しました。
https://qiita.com/release-notes/2024-07-09

今回はルールの緩和を行いましたが、機能のオン・オフについては引き続き検討していきたいと思います！
他にもお気づきの点があれば教えてください :bow: 

| 変更前 | 変更後 |
| :---: | :---: |
| ![変更前のスクリーンショット](https://github.com/increments/qiita-discussions/assets/54477535/f7b0ef84-b632-4539-8e0d-6d2adddee249) | ![変更後のスクリーンショット](https://github.com/increments/qiita-discussions/assets/54477535/2ee247f3-1b46-44d2-aaeb-2e08f72c89c4) |

**Reply by JunichiIto** [reply]
@P-SiZK 対応ありがとうございます！よかったですー 😄
また何か気付いた点があればコメントしますね。

**Comment by kimkim0814** [comment]
@JunichiIto 
お待たせしました。

この度、文章修正提案機能をルールごとにON/OFF 出来るようになりました。
https://qiita.com/release-notes/2024-10-10

是非お試しいただけると嬉しいです✨

![文章修正提案機能](https://github.com/user-attachments/assets/cf14b2c6-b2b4-4b05-b6f9-7c812297b74a)

**Reply by JunichiIto** [reply]
おお、速い！さっそくありがとうございます😆
あとで確認してみますね。
取り急ぎご連絡まで。

**Reply by JunichiIto** [reply]
遅くなりましたが、こちら動作確認できました。
ご対応どうもありがとうございました！

**Reply by tanaka-asami415** [reply]
こちらこそ、ご確認いただきありがとうございました！✨

**Comment by scivola** [comment]
そもそも「一文に同じ助詞が二回以上出てくると読みづらい」が間違っています。一概に言えません。
「は」の二回使用はたいがい避けたほうがよいでしょう。しかし以下の文は何が問題でしょうか。

> Ruby も Rust も R で始まる。
> Ruby と Rust と R を比べる。
> Ruby か Rust か R を学ぶ。

作文スキルのある人は「ああ機械ちゃんだから変な指摘してくるのね」「鬱陶しい」で済みますが，自信のない人は「えっ，これダメなん……」となるでしょう。
初心者に誤った導きを与える機能はデフォルトでオフにするのがよいのでは，とすら思いました。

せめて，「読みにくくしています」という断定ではなく，読みにくい可能性がある，というようなマイルドな表現に改め，自動校正の限界について解説したページへのリンクを置くといったふうにできないでしょうか。

自動校正は記事の質を上げるのに役立つと思うので，導入自体は賛成ですが，下手をすると作文スキルの向上をむしろ阻害する可能性もあると危惧します。

それから，

> 次の助詞が連続しているため、文を読みにくくしています。

というメッセージは日本語の文としてあまりこなれていないように思います。
これ，主語は何ですか。前半と後半がマッチしていない印象です（主語が書かれていないことを問題にしているのではありません）。
それから，「助詞が連続している」とありますが，「私は Qiita は役に立つと思います」の「は」は連続していますか。連続はしていませんよね。
（ウェブ検索してみると，自動校正で助詞の「連続」と表現しているものがたくさん見つかります。これ自動校正業界の**方言**なんですかね。）

**Reply by tsuzuki-takaaki** [reply]
@scivola 
ご提案ありがとうございます！

確かに「Ruby も Rust も」のように、頻繁に使われる表現に関する修正提案機能については、ご不便を感じるユーザーの方もいらっしゃるかと思いますので改善を検討します。

また、該当のエラーメッセージは、Qiita で使用しているライブラリの仕様に依存しているため、すぐに文言を修正するのは難しい状況です。
こちらは、ライブラリ側への改良提案も視野に入れて検討を進めます！

なお、利用しているライブラリはこちらになります。
https://github.com/textlint-ja/textlint-rule-no-doubled-joshi

よろしくお願いいたします🙇‍♂️
